applications/visualizer/frontend/src/components/viewers/TwoD/TwoDViewer.tsx

```diff
@@ -82,11 +82,7 @@ const TwoDViewer = () => {
 
   const hiddenNeurons = useMemo(() => {
     return getHiddenNeuronsIn2D(workspace);
-  }, [
-    Object.entries(workspace.visibilities)
-      .map(([name, data]) => `${name}-${data[ViewerType.Graph].visibility}`)
-      .join(","),
-  ]);
+  }, [Object.keys(workspace.visibilities).filter((neuronId) => workspace.visibilities[neuronId]?.Graph?.visibility === Visibility.Visible)]);
```

Comment:
I'm not sure, perhaps there is a logic behind, but shouldn't the dependencies compute the neurons that are actually not visible instead of visible? (so the one hidden)

Reply:
@aranega thanks for the notice, you are right this should get the hidden ones, i reversed it here by mistake 🫥